e2e/key/key.go

```diff
@@ -0,0 +1,235 @@
+		name       string
+		consoleOps []e2e.SingularityConsoleOp
+	}{
+		{
+			name: "newpair",
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("e2e test key; do not use"),
+				e2e.ConsoleSendLine("westley@sylabs.io"),
+				e2e.ConsoleSendLine("E2E test key; do not use"),
+				e2e.ConsoleSendLine("e2etests"),
+				e2e.ConsoleSendLine("e2etests"),
+				e2e.ConsoleSendLine("n"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		c.env.RunSingularity(
+			t,
+			e2e.WithPrivileges(false),
+			e2e.ConsoleRun(tt.consoleOps...),
+			e2e.WithCommand("key"),
+			e2e.WithArgs("newpair"),
+			e2e.ExpectExit(0),
+		)
+	}
+	//t.Run("singularityKeyExport", c.singularityKeyExport)
+}
+
+// singularityKeyExport will export a private, and public (binary and ASCII) key.
+func (c *ctx) singularityKeyExport(t *testing.T) {
+	tests := []struct {
+		name       string
+		armor      bool
+		secret     bool
+		exportPath string
+		consoleOps []e2e.SingularityConsoleOp
+	}{
+		{
+			name:       "export public binary",
+			exportPath: c.publicExportPath,
+			armor:      false,
+			secret:     false,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("0"),
+			},
+		},
+		{
+			name:       "export private binary",
+			exportPath: c.privateExportPath,
+			armor:      false,
+			secret:     true,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("0"),
+				e2e.ConsoleSendLine("e2etests"),
+			},
+		},
+		{
+			name:       "export public ascii",
+			exportPath: c.publicExportASCIIPath,
+			armor:      true,
+			secret:     false,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("0"),
+			},
+		},
+		{
+			name:       "export private ascii",
+			exportPath: c.privateExportASCIIPath,
+			armor:      true,
+			secret:     true,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("0"),
+				e2e.ConsoleSendLine("e2etests"),
+			},
+		},
+	}
+
+	prepCmd := func(exportPath string, secret, armor bool) []string {
+		cmd := []string{"export"}
+
+		if armor {
+			cmd = append(cmd, "--armor")
+		}
+		if secret {
+			cmd = append(cmd, "--secret")
+		}
+
+		cmd = append(cmd, exportPath)
+
+		return cmd
+	}
+
+	for _, tt := range tests {
+		c.env.RunSingularity(
+			t,
+			e2e.WithPrivileges(false),
+			e2e.WithCommand("key"),
+			e2e.WithArgs(prepCmd(tt.exportPath, tt.secret, tt.armor)...),
+			e2e.ConsoleRun(tt.consoleOps...),
+			e2e.ExpectExit(0),
+		)
+	}
+}
+
+// singularityKeyImport will export a private, and public (binary and ASCII) key.
+func (c *ctx) singularityKeyImport(t *testing.T) {
+	tests := []struct {
+		name       string
+		exportPath string
+		consoleOps []e2e.SingularityConsoleOp
+	}{
+		{
+			name:       "import public binary",
+			exportPath: c.publicExportPath,
+			consoleOps: []e2e.SingularityConsoleOp{},
+		},
+		{
+			name:       "import private binary",
+			exportPath: c.privateExportPath,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("e2etests"),
```

Comment:
Yes, that would be better for testing, but I'm trying to keep this PR small and simple. Because each `import` and `export` is ran twice. We could something like:

```go
// e2eKeyPassword is the password for the e2e test key, it will change as test run.
var e2eKeyPassword = "e2etests"
```

Then just set `e2eKeyPassword` to the new password in the test. _Theres probably a simpler way to do this..._

So for now, I added a couple "import fail tests" that will use the wrong password when importing them.